src/server/api.rs

```diff
@@ -0,0 +1,254 @@
+use super::{error::ApiError, Signature};
+
+use actix_web::client::Client;
+use http::StatusCode;
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use std::collections::HashMap;
+use tari_crypto::ristretto::RistrettoPublicKey;
+use tari_crypto::tari_utilities::hex::Hex;
+use uuid::Uuid;
+
+#[derive(Clone, Debug)]
+pub struct YatApi {
+    api_url: String,
+    api_key: String,
+}
+
+impl YatApi {
+    pub fn new(api_url: String, api_key: String) -> Self {
+        YatApi { api_url, api_key }
+    }
+
+    /// Register a new user
+    pub async fn user_create(
+        &self,
+        alternate_id: String,
+        password: String,
+    ) -> Result<DisplayUser, ApiError> {
+        let client = Client::default();
+        let json = json!({
+            "alternate_id": alternate_id,
+            "password": password,
+            "source": "Yat Partner API",
+        });
+
+        let mut response = client
+            .post(format!("{}/users", self.api_url))
+            .header("X-Api-Key", self.api_key.as_str())
+            .send_json(&json)
+            .await
+            .map_err(|e| ApiError::SendRequestError(e.to_string()))?;
+
+        // todo: error messages should be improved
+        let user: User = match response.status() {
+            StatusCode::OK => response.json().await?,
+            status => {
+                let err: ResponseError = response.json().await?;
+                log::error!("User create error ({}): {:?}", status, err);
+                return Err(ApiError::HttpError(status.to_string()));
+            }
+        };
+
+        Ok(user.user)
+    }
+
+    /// Activate a given user
+    pub async fn user_activate(
+        &self,
+        user_id: Uuid,
+        activation_source: Option<String>,
+    ) -> Result<DisplayUser, ApiError> {
+        let client = Client::default();
+        let json = json!({
+            "activation_source": activation_source,
+        });
+
+        let mut response = client
+            .patch(format!("{}/users/{}/activate", self.api_url, user_id))
+            .header("X-Api-Key", self.api_key.as_str())
+            .send_json(&json)
+            .await
+            .map_err(|e| ApiError::SendRequestError(e.to_string()))?;
+
+        let user: DisplayUser = match response.status() {
+            StatusCode::OK => response.json().await?,
+            status => return Err(ApiError::HttpError(status.to_string())),
+        };
+
+        Ok(user)
+    }
+
+    /// Log in user via alternate_id and password
+    pub async fn user_login(
+        &self,
+        alternate_id: String,
+        password: String,
+    ) -> Result<Auth, ApiError> {
+        let client = Client::default();
+        let json = json!({
+            "alternate_id": alternate_id,
+            "password": password,
+        });
+
+        let mut response = client
+            .post(format!("{}/auth/token", self.api_url))
+            .header("X-Api-Key", self.api_key.as_str())
```

Comment:
I think the reason you're getting a token for non-authenticated users is the API key is logging you in, we need to verify that this isn't the case.
```suggestion
```

Reply:
Yes you are right, just tested in Postman with no auth headers and got this error on a new/unactivated user

```
{
    "error": "Unauthorized: Email or password incorrect"
}
```

Reply:
Removed the api key header on this call

Reply:
Tested and the `process` endpoint still works correctly